### ISSUE TITLE
Option to hide keyboard

### ISSUE THREAD

**Issue by hasanelfalakiy** [Julow/Unexpected-Keyboard#607]
please add an option to hide the keyboard, as it will use a physical keyboard

**Comment by Spike-from-NH** [comment]
I think Android and its apps use commands to show or hide the soft keyboard—and that they hide it if a physical keyboard is attached.

**Comment by hasanelfalakiy** [comment]
> I think Android and its apps use commands to show or hide the soft keyboard—and that they hide it if a physical keyboard is attached.

oh is that the latest android feature? I just knew that

**Comment by Julow** [comment]
@Spike-from-NH I was not aware of this feature. Did you try it?

**Comment by hasanelfalakiy** [comment]
> @Spike-from-NH I was not aware of this feature. Did you try it?

yes, I just tried it, if when using gboard and a physical keyboard is connected, gboard is automatically hidden, but if when using an unexpected keyboard and a physical keyboard is connected, it still appears and can only be hidden if it says "done" in the enter section, do you think you can make an option to hide the keyboard? If so, I will open this issue again

**Comment by hasanelfalakiy** [comment]
> @Spike-from-NH I was not aware of this feature. Did you try it?

sir, I just tried it, if when using gboard and a physical keyboard is connected, gboard is automatically hidden, but if when using an unexpected keyboard and a physical keyboard is connected, it still appears and can only be hidden if it says "done" in the enter section, do you think you can make an option to hide the keyboard? If so, I will open this issue again

**Comment by Spike-from-NH** [comment]
Sorry, my observation may be from my era of using Hacker's Keyboard. I just tried it now.
① Edit a document (with Fossify File Manager in Android 8). When I tap a location in the text, Unexpected comes up; good.
② Plug a keyboard into the USB. Unexpected hides; good.
➌ Tap a location in the text again. Unexpected shows; that's a bug.

I don't think it should be a command the user has to invoke. Unexpected should just go away and not come back if a hard keyboard is attached. I would expect Unexpected not to reappear when the hard keyboard is unplugged, but only appear the next time it's requested. Yes, reopen the issue.

**Comment by hasanelfalakiy** [comment]
> Sorry, my observation may be from my era of using Hacker's Keyboard. I just tried it now.
> ① Edit a document (with Fossify File Manager in Android 8). When I tap a location in the text, Unexpected comes up; good.
> ② Plug a keyboard into the USB. Unexpected hides; good.
> ➌ Tap a location in the text again. Unexpected shows; that's a bug.
> 
> I don't think it should be a command the user has to invoke. Unexpected should just go away and not come back if a hard keyboard is attached. I would expect Unexpected not to reappear when the hard keyboard is unplugged, but only appear the next time it's requested. Yes, reopen the issue.

oke I agree

**Comment by Spike-from-NH** [comment]
Related? In some apps on some Android versions, if Unexpected is showing and I rotate the device, Unexpected hides; I have to tap the editing area to make Unexpected show again.

**Comment by hasanelfalakiy** [comment]
> > @Spike-from-NH I was not aware of this feature. Did you try it?
> 
> sir, I just tried it, if when using gboard and a physical keyboard is connected, gboard is automatically hidden, but if when using an unexpected keyboard and a physical keyboard is connected, it still appears and can only be hidden if it says "done" in the enter section, do you think you can make an option to hide the keyboard? If so, I will open this issue again

@Spike-from-NH  as I said here, on my device, unexpectedly it wouldn't hide when a physical keyboard was connected, and had to press "done" in the enter section if there was one, but when I use AndroidIDE, Visual Space Code or CosmicIDE the words done are not there so I still can't hide the unexpected keyboard
